src/Hint/Typecheck.hs

```diff
@@ -37,17 +37,30 @@ kindOf type_expr =
        -- kind of errors
        failOnParseError parseType type_expr
        --
-       kind <- mayFail $ runGhc1 typeKind type_expr
+       (_, kind) <- mayFail $ runGhc1 typeKind type_expr
        --
        kindToString (Compat.Kind kind)
 
+-- | Returns a string representation of the normalized type expression.
+-- This is what @:kind!@ GHCi command prints after @=@.
+normalizeType :: MonadInterpreter m => String -> m String
```

Comment:
While `:kind` prints the kind of a type, `:kind!` (with an extra `!` at the end) also normalizes the type and prints it after an equals sign:

    > :set -XTypeFamilies
    > :{
    | type family Foo a where
    |   Foo () = Int
    | :}
    > :kind Foo
    Foo :: * -> *
    > :kind (Foo ())
    (Foo ()) :: *
    > :kind! (Foo ())
    (Foo ()) :: *
    = Int